core/domain/user_services.py

```diff
@@ -2313,14 +2313,23 @@ def _get_checkpoints_in_order(init_state_name, states):
         if current_state_name not in visited_state_names:
             visited_state_names.append(current_state_name)
             current_state = states[current_state_name]
+            # We treat the last state card also as a checkpoint to store
```

Comment:
Actually I think it was intentional to omit this (I had thought about it originally). What's the point of restoring to the last state? There would be nothing else to play.

Reply:
Updated, we will not treat the last state as a checkpoint because as soon as the exploration is completed we clear the learner's progress and restart the exploration.